docs/SQLModelingStandard.md

```diff
@@ -0,0 +1,125 @@
+
+### Tables
+
+- The table name should be in **snake_case**.  
+  - The table name should be in **lowercase**.
+  - The table name made of **multiple word** should use an **underscore** 
+  in order to separate each of them.
+- The table name should be in **singular form**.
+- The table name should contain **English words** only.
+- The table name should **not contain abbreviation** with the exception of 
+  "\_rel" for entity relationship tables.
+- The table name should be a **noun** or **group of nouns**, as concise as 
+  possible.
+- When a **single word is not sufficient** to describe a table, the ordering 
+  of the words in the name should reveal the concern of the table in a 
+  broad-to-specific manner. 
+  - *i.e. A table containing the consent information of a candidate would be 
+    named `candidate_consent`, candidate being the broader concern and just 
+    `consent` is not sufficiently  informative to rule out user consents and 
+    examiner consents. A table containing the consent types would be called 
+    as such `consent_type`.*
+- A table that is **specific to a module** should be **prefixed** with the 
+  module name.
+  - *i.e. A table named `xyz` used in module `abc` should be name `abc_xyz`.*
+- **Acronyms** should be avoided, however may be used when the acronym would 
+  be **commonly used by the neuroimaging community**. The acronym should
+  **only contain upper case letter**. For historic reasons, “PSC” to refer to 
+  a Site is an exception to this rule.
+- When creating a table representing a **relation between two or more tables**. 
+  Regardless of the relation type (many-to-many, one-to-many, many-to-one or 
+  one-to-one), the table should have a composite name as such 
+  `table1_table2_rel`.
+  - *i.e. the table mapping users to their addresses would be named 
+  `user_address_rel` where `user` is the name of the Users' entity table and 
+  `address` is the name of the Addresses' entity table.*
+
+### Fields
+
+- The field name should be in **CapitalizedWords**.
+  - The field name should **start** with an **Upper case letter**.
+  - The field name should **not contain any underscore**.
+- The field name should be in **singular form**.
+- The field name should contain **English words** only.
+- The field name should **not contain abbreviations**.
+- **Acronyms** should be avoided, however may be used when the acronym would 
+  be **commonly used by the neuroimaging community**. The acronym should 
+  **only contain upper case letter**.
+- The **primary key** field of a table should be named `<TableName>ID`. While 
+  the table name is snake_case, the field names referencing it (such as the 
+  primary key) should follow field naming conventions. (ie. `my_table`’s 
+  primary key should be `MyTableID`).
+  - *i.e. the name of the primary identifier of the `country` table should be 
+  `CountryID`.*
+- When an explicit auto-incremented primary key field is present, it should be
```

Comment:
Auto_increment primary key should be avoid. While it is convenient, it doesn't represent the unicity of the rows.

Reply:
In some case, you need an artificial primary key. Composite key made of multiple fields are bad as foreign key and even without need to reference, some table have no other way.

It might not represent what is in the row, but it definitively identify a unique row.

Reply:
I am not saying that you should never do it. Just that it should not be the default option. 
There is no harm in adding an auto-increment field that will be referenced by other tables.

Reply:
This line doesn't say it should be the default option, just that **when** it is present, it should be unsigned int with auto-increment.